huaweicloud/services/aom/resource_hauweicloud_aom_prometheus_instance.go

```diff
@@ -0,0 +1,149 @@
+package aom
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/internal/entity"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/internal/httpclient_go"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+)
+
+func ResourcePrometheusInstance() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourcePrometheusInstanceRead,
+		CreateContext: resourcePrometheusInstancePatch,
+		DeleteContext: resourcePrometheusInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 100),
+					validation.StringMatch(regexp.MustCompile(
+						"^[\u4e00-\u9fa5A-Za-z0-9]([\u4e00-\u9fa5-_A-Za-z0-9]*[\u4e00-\u9fa5A-Za-z0-9])?$"),
+						"The name can only consist of letters, digits, underscores (_),"+
+							" hyphens (-) and chinese characters, and it must start and end with letters,"+
+							" digits or chinese characters."),
+				),
+			},
+			"prom_for_cloud_service": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+					"ces_metric_namespaces": {
+						Type:     schema.TypeList,
+						Optional: true,
+						Elem:     &schema.Schema{Type: schema.TypeString},
+					},
+				}},
+			},
+
+			"action": {
```

Comment:
if `action` is fixed to **prom_for_cloud_service**, we can delete this parameter